src/example_2/hardware/diffbot_system.cpp

```diff
@@ -182,38 +194,60 @@ hardware_interface::CallbackReturn DiffBotSystemHardware::on_deactivate(
   }
   // END: This part here is for exemplary purposes - Please do not copy to your production code
 
-  RCLCPP_INFO(rclcpp::get_logger("DiffBotSystemHardware"), "Successfully deactivated!");
-
-  return hardware_interface::CallbackReturn::SUCCESS;
+  if (try_to_reset_wheels() == hardware_interface::return_type::OK)
+  {
+    RCLCPP_INFO(rclcpp::get_logger("DiffBotSystemHardware"), "Successfully deactivated!");
+    return hardware_interface::CallbackReturn::SUCCESS;
+  }
+  else
+  {
+    RCLCPP_ERROR(rclcpp::get_logger("DiffBotSystemHardware"), "Couldn't deactivate!");
+    return hardware_interface::CallbackReturn::ERROR;
+  }
 }
 
 hardware_interface::return_type DiffBotSystemHardware::read(
   const rclcpp::Time & /*time*/, const rclcpp::Duration & period)
 {
   // BEGIN: This part here is for exemplary purposes - Please do not copy to your production code
-  for (std::size_t i = 0; i < hw_velocities_.size(); i++)
+  for (std::size_t i = 0; i < wheels.size(); i++)
   {
     // Simulate DiffBot wheels's movement as a first-order system
     // Update the joint status: this is a revolute joint without any limit.
     // Simply integrates
-    hw_positions_[i] = hw_positions_[i] + period.seconds() * hw_velocities_[i];
+    wheels[i].position = wheels[i].position + period.seconds() * wheels[i].velocity;
 
     RCLCPP_INFO(
       rclcpp::get_logger("DiffBotSystemHardware"),
-      "Got position state %.5f and velocity state %.5f for '%s'!", hw_positions_[i],
-      hw_velocities_[i], info_.joints[i].name.c_str());
+      "Got position state %.5f and velocity state %.5f for '%s'!", wheels[i].position,
+      wheels[i].velocity, info_.joints[i].name.c_str());
   }
   // END: This part here is for exemplary purposes - Please do not copy to your production code
 
   return hardware_interface::return_type::OK;
 }
 
-void ros2_control_demo_example_2::DiffBotSystemHardware::set_pwm_wheel_speed(
+hardware_interface::return_type ros2_control_demo_example_2::DiffBotSystemHardware::set_pwm_wheel_speed(
   int channel, double angular_speed)
 {
   // Between -1 and 1 m/s by definition in diffbot_controllers.yaml.
   double linear_wheel_speed = std::clamp(angular_speed * this->wheel_radius_, -1.0, 1.0);
 
+  // HACK ALERT!!!!!!!!!!!!!!!
```

Comment:
Good to mark down this section somewhere, memorize line 236

Reply:
Do you mean "memoize"? We could leave that for a future optimization. Might be more trouble than its worth.

Reply:
No just keep track the exact location of it in case of a quick drive train edit or something similar, not particularly storing for later use w.r.t memoization